tools/ci-build/changelogger/src/render.rs

```diff
@@ -131,10 +131,11 @@ fn date_based_release_metadata(
     ReleaseMetadata {
         title: date_title(&now),
         tag: format!(
-            "release-{year}-{month:02}-{day:02}",
+            "release-{year}-{month:02}-{day:02}-{epoch}",
```

Comment:
Is there anyway to make only releases after the first include an epoch? I expect it's infrequent to release twice in a day so it seems useful to keep the simple format as the default and only tack on the epoch if necessary

Reply:
I see the argument. We'd have to have some sort of hack in place (like the one we already have in the release pipeline where if there is already an existing tag, then append a sequential number. We are currently cloning the `aws-sdk-rust` repo with depth 20 to support this hack). We'd have to have the same hack in the `smithy-rs` repository as well to support what's suggested.

Rationale for this approach is the following:
- We don't need hack anymore, not in the smithy-rs or in the release pipeline to append a sequential number.
- I'd assume people usually see the release _TITLE_ in the release pages like `October 9th, 2024`. That won't be affected by this PR (I don't recall the last time I had to read the release tags to do something, so I assumed it was acceptable if it included an epoch suffix in it). 
- Even with the change, the release tags can be sorted by chronological and people can just read "the date portion" of the release tags, if they wish to human-read the release tags.

I'll sleep on it.